src/verify_cert.rs

```diff
@@ -1113,58 +1113,4 @@ mod tests {
                 .unwrap(),
         )
     }
-
```

Comment:
FYI, we usually rebase-merge instead of squashing so we can retain more easily reviewable atomic commits, but I don't love the commit history here. This is moving code that was just introduced in the previous commit to a different place, and splicing it into a bunch of tests that's moved into a module in the same commit.

Ideally we'd have the first commit move the tests into the module, and the second commit adding the new test.

Reply:
I'm happy to squash commits however you like. I'm used to working in repos where PRs are squash merged, sorry!

Reply:
No problem, I'm aware that the tokio repos usually squash, just making it explicit -- hopefully it won't be too painful.

Reply:
Okay, I've rewritten the history into three commits: one that adds the new tests, one that removes the tests for CNs in name constraints, and one that removes the actual common name parsing. I've also addressed your feedback on the change to `NameIterator`. Let me know what you think --- I'm happy to make additional changes to the Git history if there's anything else you'd like changed.